generate-new-strings-pot.sh

```diff
@@ -36,36 +26,55 @@ elif [[ "$BRANCH" == "HEAD" ]]; then
 	exit 1
 fi
 
-# bail if msgcat is not installed
-if ! type "msgcat" &> /dev/null; then
-	exit 1
-fi
+function join_by { local d=$1; shift; echo -n "$1"; shift; printf "%s" "${@/#/$d}"; }
+
+# All files changed in this branch
+CHANGED_FILES=$(git diff --name-only $(git merge-base $BRANCH master) $BRANCH '*.js' '*.jsx')
+
+# Diff commits to this branch
+COMMITS_HASHES=$(git log master..$BRANCH --pretty=format:%H);
+
+# Concatenate
+COMMITS_HASHES=$(join_by '\|^' ${COMMITS_HASHES[@]})
+
+# Output our json file
+printf "{\n" > localci-changed-files.json
+for file in $CHANGED_FILES; do
+	# No need to blame on a removed file
+	if [ ! -f "$(pwd)/$file" ]; then
+		break
+	fi
+	# Get all the lines that changed in our commits
+	LINES=$(git blame -flsp ${file} | grep "^${COMMITS_HASHES}" | cut -f 2 -d " ")
+	if [ -n "$LINES" ]; then
+		printf '"%s":[' "$file" >> localci-changed-files.json
+		lastline=
+		for line in $LINES ; do
+			# Also add previous line, for cases where 'translate' is on one line, and the actual string on the next
+			[[ "$lastline" -ne "$((line-1))" ]] && printf '%d,' $((line-1)) >> localci-changed-files.json
+			printf '%d,' $line >> localci-changed-files.json
+			lastline=$line
+		done;
+		sed -i '' '$ s/,$/],/' localci-changed-files.json # remove last comma
+	fi;
+done;
+sed -i '' '$ s/,$//' localci-changed-files.json # remove last comma
+printf '}\n' >> localci-changed-files.json
 
 # if node is installed, d/l node gettext tools and run
 if type "node" &> /dev/null; then
-	cd gp-localci-client/i18n-calypso
```

Comment:
At least in my local environment this failed, so I used the path of the running script.